Gemfile

```
@@ -1,5 +1,9 @@
 source 'https://rubygems.org'
 
+# mention puppet here and in the gemspec, so we can provide a specific version in CI
+
+gem "puppet", ENV["PUPPET_VERSION"]
```

I think this always overrides the version constraint from gemspec. This works in my local testing:
```suggestion
gem 'puppet', ENV.fetch('PUPPET_VERSION', ['>= 7', '< 9'])
```